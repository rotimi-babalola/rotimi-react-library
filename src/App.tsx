import React from 'react';
import styled from 'styled-components';

import Form from './components/Form';

const StyledDiv = styled.div`
  margin: auto;
  width: 50%;
  font-family: 'Roboto', sans-serif;
`;

const App = () => (
  <StyledDiv>
    <h1 style={{ color: 'palevioletred' }}>Create Account</h1>
    <Form />
  </StyledDiv>
);

export default App;
