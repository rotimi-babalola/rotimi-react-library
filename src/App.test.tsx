import { cleanup, render } from '@testing-library/react';
import React from 'react';
import App from './App';

afterEach(cleanup);

describe('App', () => {
  it('should render', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
