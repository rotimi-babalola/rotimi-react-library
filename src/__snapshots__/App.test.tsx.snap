// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<DocumentFragment>
  <div
    class="sc-AxhUy dMDaXr"
  >
    <h1
      style="color: palevioletred;"
    >
      Create Account
    </h1>
    <form
      data-testid="submit-form"
    >
      <div
        class="sc-AxhCb dWnWvi"
      >
        <label
          class="sc-AxjAm kfBTCj"
          for="username"
        >
          Username
        </label>
        <input
          class="sc-AxirZ jHZhCa"
          data-testid="username"
          name="username"
          placeholder="Type your username"
          required=""
          type="text"
        />
      </div>
      <div
        class="sc-AxhCb dWnWvi"
      >
        <label
          class="sc-AxjAm kfBTCj"
          for="password"
        >
          Password
        </label>
        <input
          class="sc-AxirZ jHZhCa"
          data-testid="password"
          name="passwprd"
          placeholder="Password"
          required=""
          type="password"
        />
      </div>
      <div
        class="sc-AxhCb dWnWvi"
      >
        <label
          class="sc-AxjAm kfBTCj"
          for="email"
        >
          Email
        </label>
        <input
          class="sc-AxirZ jHZhCa"
          data-testid="email"
          name="email"
          placeholder="Enter your email"
          required=""
          type="email"
        />
      </div>
      <button
        class="sc-AxiKw itEKzZ"
        data-testid="submit-button"
        type="submit"
      >
        Sign in
      </button>
    </form>
  </div>
</DocumentFragment>
`;
